pkg/api/norman/customization/cluster/validator.go

```diff
@@ -534,3 +538,173 @@ func validateEKS(prevCluster, newCluster map[string]interface{}) error {
 	}
 	return nil
 }
+
+func (v *Validator) validateGKEConfig(request *types.APIContext, cluster map[string]interface{}, clusterSpec *v32.ClusterSpec) error {
+	gkeConfig, ok := cluster["gkeConfig"].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	var prevCluster *v3.Cluster
+
+	if request.Method == http.MethodPut {
+		var err error
+		prevCluster, err = v.ClusterLister.Get("", request.ID)
+		if err != nil {
+			return err
+		}
+	}
+
+	// check user's access to cloud credential
+	googleCredential, ok := gkeConfig["googleCredentialSecret"].(string)
+	if !ok {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "must provide googleCredentialSecret")
+	}
+	if err := validateGKECredentialAuth(request, googleCredential, prevCluster); err != nil {
+		return err
+	}
```

Comment:
Why would this be different from the EKS credential? https://github.com/rancher/rancher/blob/1031efc0fcf9078037c841c4d8bcc3af92b78360/pkg/api/norman/customization/cluster/validator.go#L355-L360

I could move it down past line 582 to validate for creates only but I think the credential still needs to be present and validated in the import case?

Reply:
In the EKS example nothing is happening if ok is false. In your code an error is being returned. I think I was incorrect about the non-import part, both need it but only on a POST request. It is important to consider that when a user is just updating a single field, that is the only field that will be sent instead of sending the entire cluster/config.

Reply:
Got it, fixed by moving it down

Reply:
The difference is that in EKS if ok is false, then it is not checked. I was incorrect on the import part. We need to validate POST requests, be them for an import or non import cluster, or any PUT requests that are changing the actual field. It is important to note that when updating something in the UI that only the field(s) that are being changed may be sent.

Reply:
Changed this to look like the EKS validator, it should now activate on either POST or PUT but only if the field is actually present in the config.